runner/src/artifacts.rs

```diff
@@ -388,3 +458,71 @@ pub fn list_artifacts(args: &ArtifactArgs) -> ExitCode {
 
     ExitCode::SUCCESS
 }
+
+// ————————————————————————————— Process disk image ————————————————————————————— //
+
+pub fn download_disk_image(name: &str, url_path: &str) -> PathBuf {
+    let file_name: &str = extract_file_name(url_path);
+
+    if !is_file_present(&format!("artifacts/{}-miralis.img", name)) {
+        // Download image
+        log::info!("Disk image not found. Fetching the image...");
+        let mut image_downloader = Command::new("wget");
+        image_downloader.arg(url_path);
+        image_downloader
+            .status()
+            .expect("Failed to downlaod Ubuntu");
+
+        // Extract image
+        log::info!("Extracting the disk image");
+
+        let mut image_extractor;
+        let compression_type = extract_file_extension(file_name);
+        if compression_type == XZ_COMPRESSION {
+            image_extractor = Command::new("xz");
+            image_extractor.arg("-dk").arg(file_name);
+        } else if compression_type == ZST_COMPRESSION {
+            image_extractor = Command::new("zstd");
+            image_extractor.arg("-d").arg(file_name);
+        } else {
+            todo!("Implement extraction for the compression type")
+        }
+
+        image_extractor
+            .status()
+            .expect("Failed to downlaod Disk image");
+
+        // Handle case where it is .tar.smth
+        let filename_without_extension = remove_file_extention(file_name);
+        if filename_without_extension.contains(".tar") {
+            todo!("Implement support for extracting files of type .tar.something");
+        }
+
+        // Move and clean
+        log::info!("Moving and cleaning the download");
+        let mut mover = Command::new("mv");
+        mover
+            .arg(filename_without_extension)
+            .arg(format!("artifacts/{}-miralis.img", name));
+        mover.status().expect("Failed to downlaod Disk image");
```

Comment:
There is a Rust function for that in the standard Libary, let's use that instead:
https://doc.rust-lang.org/std/fs/fn.rename.html